### PR TITLE
Cellranger-atac module

### PR DESCRIPTION
**Pull request by heylf**
## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [X] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [X] Remove all TODO statements.
- [X] Emit the `versions.yml` file.
- [X] Follow the naming conventions.
- [X] Follow the parameters requirements.
- [X] Follow the input/output options guidelines.
- [X] Add a resource `label`
- [X] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

**Comment by heylf** [comment]
Dataset bundle PR here: https://github.com/nf-core/test-datasets/pull/687

**Comment by apeltzer** [comment]
@nf-core-bot fix linting

**Comment by heylf** [comment]
I copied the exception handling for conda from the cellranger modules. Not sure if there is a better way.

**Comment by klkeys** [comment]
test data from https://github.com/nf-core/test-datasets/pull/847 should be ready to use here

**Review by klkeys — COMMENTED** [review]
at a glance, some to-dos here:
* point the module to a working Docker image, might need to make one available at https://hub.docker.com/u/nfcore
* modify `tests/config/test_data.config` to point to the newly available [test data](https://github.com/nf-core/test-datasets/tree/modules/data/genomics/homo_sapiens/10xgenomics/cellranger-atac)
* wire module tests to the new test data
* pay attention to the nf-core module lint checks, the `meta.yml` files should use `directory` instead of `folder`

**Comment by heylf** [comment]
@klkeys only issues with conda now

File: modules/nf-core/cellrangeratac/mkref/main.nf [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    """
+    cellranger-atac \\
+        mkref \\
+        --config=$reference_config 2> /dev/null
```
Comment (klkeys) [inline modules/nf-core/cellrangeratac/mkref/main.nf:31]: haven't seen the redirect `2> /dev/null` line before, what's the rationale?

Comment (klkeys) [reply modules/nf-core/cellrangeratac/mkref/main.nf:31]:
```suggestion
        --config=$reference_config \\
        $args
```
regardless of `2> /dev/null` this module is missing an `$args` option

File: tests/config/pytest_modules.yml [hunk 2]
```diff
@@ -666,6 +666,20 @@ cellranger/vdj:
   - modules/nf-core/cellranger/vdj/**
   - tests/modules/nf-core/cellranger/vdj/**
 
+cellrangeratac/count:
+  - modules/nf-core/cellrangeratac/count/**
+  - tests/modules/nf-core/cellrangeratac/count/**
+  - modules/nf-core/cellrangeratac/mkref/**
+  - tests/modules/nf-core/cellrangeratac/mkref/**
+
```
Comment (klkeys) [inline tests/config/pytest_modules.yml:674]:
```suggestion
  - modules/nf-core/unzip/**
  - tests/modules/nf-core/unzip/**
  
```
do you need to add this? normally all used modules are included, and your `cellrangeratac/count` module uses `UNZIP`

File: tests/config/pytest_modules.yml [hunk 3]
```diff
@@ -666,6 +666,20 @@ cellranger/vdj:
...
+cellrangeratac/mkfastq:
+  - modules/nf-core/cellrangeratac/mkfastq/**
+  - tests/modules/nf-core/cellrangeratac/mkfastq/**
+
```
Comment (klkeys) [inline tests/config/pytest_modules.yml:678]:
```suggestion
  - modules/nf-core/unzip/**
  - tests/modules/nf-core/unzip/**
  
```

**Review by klkeys — CHANGES_REQUESTED** [review]
regarding conda, you need to edit the Github Actions workflow to exclude the conda pull

see example [here](https://github.com/nf-core/modules/blob/e8d164fa8326069de538f74910d809da9f25826e/.github/workflows/pytest-workflow.yml#L72) from the `cellranger` module bundle

you can probably just append similar lines for the three `cellranger-atac` modules in this PR

**Comment by heylf** [comment]
Still discussion docker images

**Comment by klkeys** [comment]
oh dear, what's the reason for closing this?

**Comment by heylf** [comment]
Sorry, this was a bad mistake of mine. I reopened a new PR with all the changes you have mentoned: https://github.com/nf-core/modules/pull/3575

File: tests/config/pytest_modules.yml (hunk 2 above)
Comment (heylf) [reply tests/config/pytest_modules.yml:674]: No you dont have to. It works without and other tools which use UNZIP aso do not include it in the pytest_modules.yml.

**Pull request closed without merging**